Code/GraphMol/Atropisomers.cpp

```diff
@@ -538,19 +538,35 @@ void detectAtropisomerChirality(ROMol &mol, const Conformer *conf) {
     }
   }
 
+  bool needsHybridization = false;
   for (auto bondToTry : bondsToTry) {
-    if (bondToTry->getBeginAtom()->needsUpdatePropertyCache()) {
-      bondToTry->getBeginAtom()->updatePropertyCache(false);
+    // defer cache update on the whole mol unless we actually have bonds to try
+    // we need to do an update on the whole mol and not just incident atoms
+    // because we need to calculate hybridization, which is non-local
+    for (const auto atom : mol.atoms()) {
+      if (atom->getAtomicNum() == 0) {
+        continue;  // dummy atoms stay unspecified
+      }
+      if (atom->getHybridization() == Atom::HybridizationType::UNSPECIFIED) {
+        needsHybridization = true;
+        break;
+      }
     }
-    if (bondToTry->getEndAtom()->needsUpdatePropertyCache()) {
-      bondToTry->getEndAtom()->updatePropertyCache(false);
+    if (mol.needsUpdatePropertyCache() || needsHybridization) {
```

Comment:
Once the search for atoms that need hybridization assigned has been moved out of the loop, this can go directly underneath it.